### PR TITLE
User can register for an account to view 5 free articles API

### PR DESCRIPTION
**Pull request by ViktorHek**
[User can register for an account to view 5 free articles API](https://www.pivotaltracker.com/story/show/176031765)
### User Story
```
As an API,
In order for users to access 5 free articles,
We would like to authenticate the visitor.
```
## Changes proposed in this pull request:
* adding functionality for registered user to read 5 free articles, then they have to subscribe.
* writing test to check it registered user can read 5 free articles and no more
* adding column article_click to user
## What I have learned working on this feature:
* what `and return` means
* how to count clicks in an API
* expect(response_json).to have_key('xyz').and have_value('xyz') in spec

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -37,7 +38,11 @@ def is_user_journalist?
   end
 
   def is_user_authorized_article?
-    render json: { message: 'You are not catscribed yet? You shall be' }, status: 401 unless current_user.subscriber? || current_user.journalist?
+    unless current_user.subscriber? ||
+       current_user.journalist? || 
+       current_user.article_click < 5
```
Comment (Ehugo2000) [inline app/controllers/api/articles_controller.rb:43]: nice!

File: spec/requests/api/registered_user_can_read_five_free_articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  describe 'registered user can successfully' do
+    let(:registered_user) { create(:registered_user) }
+    let(:registered_user_headers) { registered_user.create_new_auth_token }
+    describe 'read first free article' do
+      before do
+        get "/api/articles/#{article.id}",
+            headers: registered_user_headers
+      end
+
```
Comment (Ehugo2000) [inline spec/requests/api/registered_user_can_read_five_free_articles_spec.rb:12]: Format this test

File: spec/requests/api/subscriber_can_see_specific_article_spec.rb [hunk 3]
```diff
@@ -41,14 +43,13 @@
...
-      get "/api/articles/#{article.id}",
-          headers: registered_user_headers
+      get "/api/articles/#{article.id}"
     end
     it 'is expected to return a error status' do
       expect(response).to have_http_status 401
     end
     it 'is expected to respond with a error message' do
-      expect(response_json['message']).to eq 'You are not catscribed yet? You shall be'
+      expect(response_json['errors'][0]).to eq "You need to sign in or sign up before continuing."
```
Comment (Ehugo2000) [inline spec/requests/api/subscriber_can_see_specific_article_spec.rb]: change to have_key

**Review by Ehugo2000 — COMMENTED** [review]
Nice work, only formatting tests

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (ViktorHek) [reply app/controllers/api/articles_controller.rb:43]: thanks :)

File: spec/requests/api/registered_user_can_read_five_free_articles_spec.rb (hunk 2 above)
Comment (ViktorHek) [reply spec/requests/api/registered_user_can_read_five_free_articles_spec.rb:12]: done :)

File: spec/requests/api/subscriber_can_see_specific_article_spec.rb (hunk 3 above)
Comment (ViktorHek) [reply spec/requests/api/subscriber_can_see_specific_article_spec.rb]: done :)

**Pull request merged**